ファイル一覧表示用のアクションの追加

a5ob7r opened the pull request:
# 概要

APIサーバの `/files` リソースにGETリクエストを投げてレスポンスに含まれているアップロード済みのファイル一覧のメタデータを表示する。

その際のレスポンスには、ヘッダーには、総ファイル件数や現在のページ番号のような、ページネーションに関するメタデータが含まれているため、それらの情報を利用して、ページネーション用のリンクや現在のページに含まれてるファイルの位置情報などを表示する。

なお、HTMLには必要な情報が表示されていれば問題ない程度の認識で書いてあるので、大変大雑把な内容になっている。

a5ob7r committed:
Add the main logic to the index files action

a5ob7r committed:
Add the view for the index files action

a5ob7r committed:
Add the spec for the files index action

a5ob7r committed:
Add a more accurate error handling

a5ob7r committed:
Change the variable name to resolve the name conflict

a5ob7r committed:
Extract redundant login steps as a before block

a5ob7r committed:
Add the pagination to the index files action

a5ob7r committed:
Use a more specific error object to handle errors

tk0miya commented on Gemfile:
```diff
@@ -9,6 +9,7 @@ gem 'rails', '~> 7.0.6'
 
 gem 'activerecord-session_store', '~> 2.0'
 gem 'flexirest', '~> 1.11'
+gem 'link-header-parser', '~> 5.0'
```
メモ: https://github.com/thlorenz/parse-link-header

tk0miya commented on app/helpers/pagination_helper.rb:
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module PaginationHelper
+  def pagination_entires_info(**metadata)
```
typo?
```suggestion
  def pagination_entries_info(**metadata)
```

a5ob7r replied:
意図してそのように書いてはいたのですが、間違いでした。

kaminariのhelperメソッド名をいじってそのような名前にしたのですが、その際に `entries` を `entires` と読んでしまっていました。

https://github.com/kaminari/kaminari#the-page_entries_info-helper-method

少しだけ違和感は感じていたのですが、見直すには至りませんでしたね...。

a5ob7r committed:
Fix the incorrect naming for pagination entries info

The original helper method name is `page_entries_info`.

https://github.com/kaminari/kaminari#the-page_entries_info-helper-method
